modules/caddyhttp/rewrite/caddyfile.go

```diff
@@ -25,20 +25,99 @@ func init() {
 
 // parseCaddyfile sets up the handler from Caddyfile tokens. Syntax:
 //
-//     rewrite [<matcher>] <to>
+//     rewrite [<matcher>] [<uri>] {
+//         uri                <string>
+//         method            <string>
+//         strip_path_prefix <string>
+//         strip_path_suffix <string>
+//         rehandle          <bool>
+//         http_redirect     <status_code>
+//     }
 //
-// The <to> parameter becomes the new URI.
 func parseCaddyfile(h httpcaddyfile.Helper) (caddyhttp.MiddlewareHandler, error) {
-	var rewr Rewrite
+	var (
+		rewr Rewrite
+
+		uriIsSet       bool
+		nextBlockIsSet bool
+		rehandleIsSet  bool
+		redirectIsSet  bool
+	)
+
 	for h.Next() {
-		if !h.NextArg() {
+		args := h.RemainingArgs()
+		switch len(args) {
+		case 0:
+		case 1:
+			uriIsSet = true
+			rewr.URI = h.Val()
+		default:
 			return nil, h.ArgErr()
 		}
-		rewr.URI = h.Val()
-		if h.NextArg() {
-			return nil, h.ArgErr()
+
+		for h.NextBlock(0) {
+			nextBlockIsSet = true
+			switch h.Val() {
+			case "uri":
+				if uriIsSet {
+					return nil, h.Err("uri is already set")
+				}
+				if !h.Args(&rewr.URI) {
+					return nil, h.ArgErr()
+				}
+			case "method":
+				if !h.Args(&rewr.Method) {
+					return nil, h.ArgErr()
+				}
+			case "strip_prefix":
+				if !h.Args(&rewr.StripPathPrefix) {
+					return nil, h.ArgErr()
+				}
+			case "strip_suffix":
+				if !h.Args(&rewr.StripPathSuffix) {
+					return nil, h.ArgErr()
```

Comment:
I had a second thought, that perhaps `strip_prefix` and `strip_suffix` should be separate directives -- especially since they would conflict with the "uri" parameter, I think.

Would that be OK with you? Making them separate directives is suuuper easy... and I think having more directives that do fewer things is good, especially for nuanced stuff like rewriting.

Reply:
Sounds reasonable to me.